src/ecmascript/spidermonkey/document.c

```diff
@@ -492,3 +552,122 @@ document_writeln(JSContext *ctx, unsigned int argc, JS::Value *rval)
 {
 	return document_write_do(ctx, argc, rval, 1);
 }
+
+// Helper function for document replace
+char *
+str_replace(char *orig, char *rep, char *with)
+{
```

Comment:
I guess it as a good place to use struct string.

Reply:
Are You sure? In this case I'm using char * because I'm working with cache fragment data. That structure is char *. I'd have to covert it back na forth. I'll review that once more. But since rewritting that means working with strimg->source most of the time I doubt it's effective.
Please provide reasoning for this request for change.
What would that bring? And is it any good to convert this low-level function into the string replacement if there is need to work with char *.
If the struct string replace function is needed it could be implemented. But right now in this case str replace using char seems a better idea for me.
If You're unhappy with this function I can see what ca I do with it.

Reply:
ok, maybe you are right. But in other places prefer string, please.

Reply:
:) ok, thank You very much. Just in time when I finished rewrite of the string replace function to struct string. I can probably provide that too. Maybe someone can use it in the future. Let me know if You're interested in the function. I can add it to util/string upon Your request.

Reply:
O.K., agreed. Fixed. You we right. It's better to use struct string. Thank You very much.